graphistry/feature_utils.py

```diff
@@ -619,11 +682,19 @@ def fit_pipeline(
     columns = X.columns
     index = X.index
 
-    X = transformer.fit_transform(X)
-    if keep_n_decimals:
-        X = np.round(X, decimals=keep_n_decimals)  #  type: ignore  # noqa
-
-    return pd.DataFrame(X, columns=columns, index=index)
+    X_type = str(getmodule(X))
+    if 'cudf' not in X_type:
+        X = transformer.fit_transform(X)
+        if keep_n_decimals:
+            X = np.round(X, decimals=keep_n_decimals)  #  type: ignore  # noqa
+        X = pd.DataFrame(X, columns=columns, index=index)
+    else:
+        X = transformer.fit_transform(X.to_numpy())
+        if keep_n_decimals:
+            X = np.round(X, decimals=keep_n_decimals)  #  type: ignore  # noqa
+        _, _, cudf = lazy_import_has_dependancy_cu_cat()
```

Comment:
`assert cudf is not None`

Reply:
good practice to assert even after if statement check? good to know, yay learning

Reply:
It's useful when you don't want the 'if' but can imagine future changes or misuses accidentally getting the assumption wrong